core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -284,7 +280,9 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
             tableFileIO(response.config()),
             response.tableMetadata());
 
-    BaseTable table = new BaseTable(ops, fullTableName(loadedIdent), this.scanReporter);
+    BaseTable table =
+        new BaseTable(
+            ops, fullTableName(loadedIdent), new RESTScanReporter(client, paths, headers(context)));
```

Comment:
Rather than having a separate class that gets passed the `RESTClient`, what about just using lambda that calls a method in `RESTSessionCatalog`?

```java
  BaseTable table = new BaseTable(ops, fullTableName(loadedIdent), report -> reportScan(report, session::headers));
  ...

  private void reportScan(ScanReport report, Supplier<Map<String, String>> headers) {
    client.post(
        paths.scanReport(),
        ScanReportRequest.builder().fromScanReport(report).build(),
        null,
        headers,
        ErrorHandlers.defaultErrorHandler());
  }
```

That leaks fewer internals of the catalog (like `ResourcePaths`) to other objects.

I'd also use `session::headers` rather than `headers(context)` because that avoids going through the session cache. We already have the `AuthSession` loaded and we also don't want to worry about session expiration.

Reply:
that makes sense, updated.